Fix libstd on DragonFly

mneumann opened the pull request:
Following changes:

* birthtime does not exist on DragonFly
* errno: __dfly_error is no more. Use #[thread_local] static errno.
* clock_gettime expects a c_ulong

These changes are required to build DragonFly snapshots again.

rust-highfive commented:
r? @aturon

(rust_highfive has picked a reviewer for you, use r? to override)

alexcrichton commented on src/libstd/sys/unix/time.rs:
```diff
@@ -317,6 +318,20 @@ mod inner {
+                libc::clock_gettime(clock as libc::c_ulong, &mut t.t)
+            }).unwrap();
+            t
+        }
```
Could the duplication here be reduced from the whole function to just perhaps a typedef?

alexcrichton replied:
Either that or the `now` here could take an argument of the appropriate type

mneumann replied:
Ideally we would export the type as ```clock_t``` from libc, then we could just use it here. But I think in this case a typedef is just fine!

mneumann commented:
@aturon @alexcrichton I rebased and squashed into a single commit. for some unknown reason the travis build shows a failed state (it successfully ran the tests).

alexcrichton commented:
@bors: r+ 60c988ec176b9653bc1c3a6d6a738646d15a5ad6
